### PR TITLE
ref: api review

### LINKED ISSUE #155: API review
This issue summarizes a number of changes with a small impact. I'll open separate issues for larger changes that may need discussion:

1. [ ] 1. Pretty print all responses by default (indent with one tab) and remove the `?pretty` param. The extra whitespace has [close to zero impact](https://www.vinaysahni.com/best-practices-for-a-pragmatic-restful-api#pretty-print-gzip) when gzip is used, but it's much easier to work with for users when everything is readable.
2. [ ] 2. Add etag and compression support, see https://github.com/jsdelivr/data.jsdelivr.com/blob/master/src/index.js#L76-L85
3. [ ] 3. Set case-sensitive and strict routing](https://github.com/jsdelivr/data.jsdelivr.com/blob/master/src/index.js#L26) (note that this doesn't affect variables which can stay case-insensitive, just endpoint names).
4. [ ] 4. Use ISO format for all time data everywhere, instead of timestamps (`date.toISOString()`).
5. [ ] 5. `X-RateLimit-Reset` header should return the number of seconds before the limit is reset, instead of a fixed date (`Math.round(response.msBeforeNext / 1000)`.
6. [ ] 6. Remove `@koa/cors`, and set the following headers manually for all responses:
   ```
   'Access-Control-Allow-Origin': '*',
   'Access-Control-Allow-Headers': '*, Authorization',
   'Access-Control-Expose-Headers': '*',
   'Cross-Origin-Resource-Policy': 'cross-origin',
   'Timing-Allow-Origin': '*',
   'Vary': 'Accept-Encoding',
   ```
   @koa/cors varies everything by Origin header and that's not desirable here.
7. [ ] 7. In response to `POST /measurements`, send status code 202 and location header with the absolute URL to the measurement, which can be used for GET later (the server name [must be configurable](https://github.com/jsdelivr/data.jsdelivr.com/blob/master/config/default.js#L5)).
8. [ ] 8. Location names should be returned by the server in a form that can be used by clients to render them directly in the UIs with the expected casing - i.e. country/region/city/network should have first letters in uppercase (for multiple word names, follow regular English rules, or rather, store the names in the DB without any lowercasing). country/continent/asn codes should be all uppercase. This is all related to responses from the server only - the matching logic can stay case-insensitive.

### PR DESCRIPTION
**Pull request by patrykcieszkowski**
resolve #155

### CODE REVIEW

**Comment by MartinKolarik** [comment]
@patrykcieszkowski let me know when all points from #155 are covered, and we can merge this.

**Comment by patrykcieszkowski** [comment]
> @patrykcieszkowski let me know when all points from #155 are covered, and we can merge this.

It's all done, with exception to the 7.1 and 7.2, following our exchange on the ticket. But I was thinking of updating docs on this PR, following changes on #157

File: docs/measurement/post-create.md [hunk 1]
```diff
@@ -78,7 +76,8 @@ for `Locations` schema, please see [LOCATION SCHEMA](./schema/location.md).
 POST https://api.globalping.io/v1/measurements/
 {
     "id": "PY5fMsREMmIq45VR",
-    "probesCount": 1
+    "probesCount": 1,
+    "uri": "https://api.globalping.io/v1/measurements/PY5fMsREMmIq45VR"
```
[the remark this answers is not in the document]
Comment (patrykcieszkowski) [reply docs/measurement/post-create.md:80]: I guess we can use `Location: <URI>` too.

**Comment by patrykcieszkowski** [comment]
> > @patrykcieszkowski let me know when all points from #155 are covered, and we can merge this.
> 
> It's all done, with exception to the 7.1 and 7.2, following our exchange on the ticket. But I was thinking of updating docs on this PR, following changes on #157

docs updates for #157 will be added to https://github.com/jsdelivr/globalping/pull/154

File: docs/measurement/post-create.md [hunk 2]
```diff
@@ -59,16 +56,16 @@ for `Locations` schema, please see [LOCATION SCHEMA](./schema/location.md).
 
 ## success response
 
-**status code**: `200 OK`
+**status code**: `202 Accepted`
 
-**content**: response will contain an Id number of your measurement, and total number of probes assigned to your query. The count of assigned probes might vary from what you requested.
+**content**: response will contain an Id number of your measurement, an absolute URI to the measurement status, and total number of probes assigned to your query. The count of assigned probes might vary from what you requested.
```
Comment (MartinKolarik) [inline docs/measurement/post-create.md:61]:
```suggestion
**content**: response will contain an Id number of your measurement, and total number of probes assigned to your query. The count of assigned probes might vary from what you requested. A URL pointing to the measurement status is sent in the `Location` header.
```

**Review by MartinKolarik — APPROVED** [review]

**Pull request merged**